scalafmt-tests/src/test/resources/test/JavaDoc.stat

```diff
@@ -1815,3 +1815,126 @@ val a = 1
   * }}}
   */
 val a = 1
+<<< #2498 format fenced scala code block, Asterisk
+docstrings.style = Asterisk
+docstrings.wrap = yes
+maxColumn = 50
+===
+/** Nice scala code:
+  *  ```scala
+  *   class Formatter(style: ScalaStyle, tree1: Tree)
+  *   class Formatter(style: ScalaStyle, tree12: Tree)
+  *   class Formatter(style: ScalaStyle, tree: Tree, toks: Array[FormatToken], statementStarts: Set[Token], owners: Map[Token, Tree])
+  * ```
+  * No more nice scala code.
+  */
+val a = 1
+>>>
+/**
+ * Nice scala code:
+ * ```scala
+ * class Formatter(style: ScalaStyle, tree1: Tree)
+ * class Formatter(
+ *   style: ScalaStyle,
+ *   tree12: Tree
+ * )
+ * class Formatter(
+ *   style: ScalaStyle,
+ *   tree: Tree,
+ *   toks: Array[FormatToken],
+ *   statementStarts: Set[Token],
+ *   owners: Map[Token, Tree]
+ * )
+ * ```
+ * No more nice scala code.
+ */
+val a = 1
+<<< #2498 format fenced scala code block, SpaceAsterisk
+docstrings.style = SpaceAsterisk
+docstrings.wrap = yes
+maxColumn = 50
+===
+/** Nice scala code:
+  *  ```scala
+  *   class Formatter(style: ScalaStyle, tree1: Tree)
+  *   class Formatter(style: ScalaStyle, tree12: Tree)
+  *   class Formatter(style: ScalaStyle, tree: Tree, toks: Array[FormatToken], statementStarts: Set[Token], owners: Map[Token, Tree])
+  * ```
+  * No more nice scala code.
+  */
+val a = 1
+>>>
+/** Nice scala code:
+  * ```scala
+  * class Formatter(
+  *   style: ScalaStyle,
+  *   tree1: Tree
+  * )
+  * class Formatter(
+  *   style: ScalaStyle,
+  *   tree12: Tree
+  * )
+  * class Formatter(
+  *   style: ScalaStyle,
+  *   tree: Tree,
+  *   toks: Array[FormatToken],
+  *   statementStarts: Set[Token],
+  *   owners: Map[Token, Tree]
+  * )
+  * ```
+  * No more nice scala code.
+  */
+val a = 1
+<<< #2498 format fenced scala code block, AsteriskSpace
+docstrings.style = AsteriskSpace
+docstrings.wrap = yes
+maxColumn = 50
+===
+/** Nice scala code:
+  *  ```scala
+  *   class Formatter(style: ScalaStyle, tree1: Tree)
+  *   class Formatter(style: ScalaStyle, tree12: Tree)
+  *   class Formatter(style: ScalaStyle, tree: Tree, toks: Array[FormatToken], statementStarts: Set[Token], owners: Map[Token, Tree])
+  * ```
+  * No more nice scala code.
+  */
+val a = 1
+>>>
+/** Nice scala code:
+ *  ```scala
+ *  class Formatter(
+ *    style: ScalaStyle,
+ *    tree1: Tree
+ *  )
+ *  class Formatter(
+ *    style: ScalaStyle,
+ *    tree12: Tree
+ *  )
+ *  class Formatter(
+ *    style: ScalaStyle,
+ *    tree: Tree,
+ *    toks: Array[FormatToken],
+ *    statementStarts: Set[Token],
+ *    owners: Map[Token, Tree]
+ *  )
+ *  ```
+ *  No more nice scala code.
+ */
+val a = 1
+<<< #2498 do not format broken fenced scala code block
+docstrings.style = SpaceAsterisk
+docstrings.wrap = yes
+===
+/** Nice scala code:
+  *   ```scala    sc:compile
+  * class Formatter(style: ScalaStyle, tree: Tree, toks: Array[FormatToken
+  * ```
+  */
+val a = 1
+>>>
+/** Nice scala code:
+  * ```scala sc:compile
```

Comment:
Hmm... this means if we could format it twice and each one would change the comment in a different way. Should we maybe diviate from the spec and just allow ``` after any number of spaces?

Reply:
this part i didn't understand 🙂 can you explain, please?

Reply:
I commented on a wrong part I think. I meant the situation in which fences are in a wrong spot (too many spaces) 
First ``` are not in a correct place:
```
/** Nice scala code:
  *    ```scala    sc:compile
  * class Formatter(style: ScalaStyle, tree: Tree, toks: Array[FormatToken])
  * ```
  */
```

after first formatting you get them into the correct place:
/** Nice scala code:
  * ```scala    sc:compile
  * class Formatter(style: ScalaStyle, tree: Tree, toks: Array[FormatToken])
  * ```
  */
```

and after second formatting you will format the code inside the fences since they are correct now:
```
/** Nice scala code:
 *  ```scala
 *  class Formatter(
 *    style: ScalaStyle,
 *    tree: Tree,
 *    toks: Array[FormatToken]
 *  )
 *  ```
 *  No more nice scala code.
 */
```

Reply:
hmm... it could happen but only under very lucky circumstances. opening quotes would have to start a line, and closing quotes would have to start a line and be the last tokens in a paragraph.

how should we fix?

Reply:
We should probably not adjust a line with spaces and ``` , because that actually changes the meaning of the comments. But again, it's not a big deal, so maybe it's fine to leave as is.

Reply:
then we need to change the parser (in scalameta) to specify how many spaces were before the fence.